lib/msal-common/src/client/DeviceCodeClient.ts

```diff
@@ -143,6 +143,7 @@ export class DeviceCodeClient extends BaseClient {
         const requestBody = this.createTokenRequestBody(request, deviceCodeResponse);
         const headers: Record<string, string> = this.createDefaultTokenRequestHeaders();
 
+        const userSpecifiedTimeout = request.timeout ? TimeUtils.nowSeconds() + request.timeout : undefined; 
```

Comment:
Shouldn't we validate the user set timeout? `request.timeout <= deviceCodeExpirationTime` should always hold. Can we throw a configuration error if this isn't respected?

Reply:
Since the `deviceCodeExpirationTime` is provided at runtime, won't throwing a configuration error be too rough of an experience in the case Azure changes their `deviceCodeExpirationTime`?

I will however still add that validation to ensure the user timeout is only respected if it's shorter than the device code expiration time

Reply:
I agree throwing an error might be too extreme here: would probably be better to just use the shortest time.

Reply:
I also agree. In the case where we go to the default we should do a verbose log explaining why the user provided timeout was ignored. Also I think this is the kind of configuration option that may warrant a note on the Device Code flow section of the Request doc in msal-common.

Reply:
Two great points @hectormmg, went ahead and updated the docs as well as added a verbose log for when the user timeout is ignored.

Reply:
Thanks. Looks good for me.